### PR TITLE
RDD: Some initial guidelines on chart development

### PR DESCRIPTION
**Pull request by yorinasub17**
After prototyping and reading some documents on how to develop helm charts, I landed on these set of guidelines for chart development.

The tl;dr is:

- Make sure you follow [the official best practices on chart development](https://docs.helm.sh/chart_best_practices/)
- Don't provide defaults in `values.yaml` for required inputs
- Use `linter_values.yaml` for required inputs compatible with `helm lint` (see https://github.com/gruntwork-io/pre-commit/#linter_valuesyaml for an explanation).
- Document values, even more than the official guide recommends.
- There are two sets of tests for every helm chart: helm tests and terratests
    - helm tests are run by helm, and is used to check the deployment. This means that these tests are meant to be run in production to validate the chart install.
    - terratests are the integration tests for the charts. These tests include installing the chart with various different inputs.

### CODE REVIEW

**Comment by yorinasub17** [comment]
`helmcharts.gruntwork.io` is live, so merging
